src/framework/audio/common/async/internal/queuepool.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore Limited and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "queuepool.h"
+
+using namespace muse::audio::comm;
+
+QueuePool* QueuePool::instance()
+{
+    static QueuePool p;
+    return &p;
+}
+
+QueuePool::QueuePool()
+{
+    m_threads.resize(MAX_THREADS);
+}
+
+void QueuePool::regPort(const std::thread::id& th, const std::shared_ptr<Port>& port)
+{
+    for (size_t i = 0; i < m_threads.size(); ++i) {
+        ThreadData& thdata = m_threads.at(i);
+        if (thdata.threadId == th) {
+            thdata.ports.push_back(port);
+            break;
+        } else if (thdata.threadId == EMPTY_THREAD_ID) {
+            thdata.threadId = th;
+            thdata.ports.push_back(port);
+            break;
+        }
+    }
+}
+
+void QueuePool::unregPort(const std::thread::id& th, const std::shared_ptr<Port>& port)
+{
+    for (size_t i = 0; i < m_threads.size(); ++i) {
+        ThreadData& thdata = m_threads.at(i);
+        if (thdata.threadId == th) {
+            thdata.ports.erase(std::remove(thdata.ports.begin(), thdata.ports.end(), port), thdata.ports.end());
+
+            if (thdata.ports.empty()) {
+                thdata.threadId = EMPTY_THREAD_ID;
```

Comment:
This means that there can be an `EMPTY_THREAD_ID` in the middle of `m_threads`, but that goes not well together with the `break;` three lines below and in `processMessages`, right?